### ISSUE TITLE
Add Samsung Gear VR Controller support

### ISSUE THREAD

**Issue by rdady** [kozec/sc-controller#399]
Of course it would be nice to use the Gear VR Controller with sc-controller!!!!
I have built a POC code for the gearvrcontroller in python3, which I would love to contribute to this project.
The controller is recognized as other device, not as a HID.
The user needs to pair and connect to PC via bluetooth, the program reads the touchpad absolute coordinates and issues uinput mouse relXY displacement. The mouse movement is kind of funny, no smoothing implemented yet.
Interested?

**Comment by kozec** [comment]
> Interested?

Definitely, although if we are talking about [this device](https://www.amazon.com/Samsung-ET-YO324BBEGUS-Gear-VR-Controller/dp/B06XHXRXP1), one may need two to cover most games :)

**Comment by rdady** [comment]
That is exactly that device.
I don't know, whether you need 2 of them, maybe. The amazon deal sells one headset with one controller.
Everyone has a different usecase, mine would be one handed mouse - browsing around while having my newborn on my shoulder and not being able to lean as close to the PC as usually. A SC is not easy to hold in one hand. I consider the gearvrcontroller being the one handed brother of the SC.

In my opinion sc-controller is sometimes used beyond gaming as well - and that is cool :)

**Comment by kozec** [comment]
> I came so far. The pointer smoothing is really a point I haven't come clear with.

There is pointer [smoothing code in SC-Controller that could be used](https://github.com/kozec/sc-controller/blob/a9731ea7a2d4856bb2f97ee630359dfcf8940677/scc/modifiers.py#L1434). It's fairly simple - just remember X last positions that HW sent and use weighted (or normal, but weighted works better) average as output.

**Comment by rdady** [comment]
Yes, I have already seen it.
The thing is, that the touchpad gives readings between 10-315 unlike a SC from -32767 to +32767.
Untouched reads as touchposition 0,0. And the refresh rate is also below the SC, which I haven't measured yet.
I will give it a try xdotools as well, instead of uinput.

**Comment by parkerlreed** [comment]
@rdady Just stumbled across your project after looking for forever (granted a few months back)

Love it!
